src/Client.php

```diff
@@ -111,16 +126,14 @@ public function runRawQuery(string $queryString, $resultsAsArray = false, array
     {
         $request = new Request($this->requestMethod, $this->endpointUrl);
 
-        foreach($this->httpHeaders as $header => $value) {
-            $request = $request->withHeader($header, $value);
-        }
-
```

Comment:
I just realized a problem while running some scenarios in my head. So currently, headers are used for multiple purposes other than just authorization, such as passing content type for instance. What is a user sets some headers for their requests in the constructor but they also want to use AWS authorization? With the current implementation, AWS authenticator will takeover and ignore all the headers the user passed for non-authorization purposes, causing regression.

In other words, because headers are used for authorization and non-authorization purposes, when we bypass setting the headers because of AWS authorization, we lose the headers passed by the user for non-authorization purposes.

To solve this problem, I propose one of two solutions:
1. Have the AwsIamAuth extend HeaderAuth, and in the `run()` implementation, execute the `parent::run()` at the beginning to set headers, then pass the returned request for the current logic.
2. Revert this change and have the for loop in all cases, remove the `HeaderAuth` class, and only run the authenticator `run()` if it's not null.

I'm open to any other options you have.

Reply:
I understand. So made changes as described in the second option.